### DISCUSSION TITLE
root password change

### DISCUSSION THREAD

**Discussion by Razriel3** [cockpit-project/cockpit#17701] [Q&A] [answered]
I have headless OS build with YOCTO.
First time I boot my system after flashing the image it requires a root password change.
I am using cockpit for linux platform managment.
Currently get the following error in the login page:
Authentication failed: internal-error: You are required to change your password immediately (administrator enforced).
my question is:
Can it be done from the login page / alternate page ?
do I need to configure the cockpit somehow  ?

**Comment by martinpitt** [comment]
In general, changing the password on the login page does work. We have an 
[integration test](https://github.com/cockpit-project/cockpit/blob/69dc4b1eaf6bc2d9c4ef164cbd4b6f8183fd6dc4/test/verify/check-static-login#L262) that checks this at least for a non-root user, on the OSes that we [support](https://cockpit-project.org/running.html#installation-setup). However, we don't test on yocto -- the PAM configuration is annoyingly different between operating systems, and it could very well be that something in the conversation isn't recognized. 

So this needs to be debugged, preferably by someone who knows about yocto. This would also need some detailed instruction steps what you did, i.e. how to reproduce the setup.

**Comment by martinpitt** [comment, marked as answer]
Issue #17760 is about that, so let's track that in one place only.
